sdk/src/client/api/block_builder/input_selection/core/mod.rs

```diff
@@ -165,11 +165,12 @@ impl InputSelection {
 
     /// Creates a new [`InputSelection`].
     pub fn new(
-        available_inputs: Vec<InputSigningData>,
-        outputs: Vec<Output>,
-        addresses: Vec<Address>,
+        available_inputs: impl Into<Vec<InputSigningData>>,
+        outputs: impl Into<Vec<Output>>,
```

Comment:
Is there a specific reason, why we use `Into<Vec<>>`  for outputs, but `IntoIterator<>` for address? `IntoIterator` seems to be more generic and universal. `Vec` is always heap-based.

Reply:
I don't think it really depends on whether it's outputs or address, but whether if at some point we really need a Vec or not. And in this specific case we're storing the Vec in a struct.

Reply:
Yeah, and `collect()` seems (not 100% sure) optimized by the compilator when you provide a vector.

Reply:
I could believe that collecting an owned vec is optimized. If that's the case, we probably should use it rather than `Into<Vec<_>>`, but I wasn't sure. Can you prove that this case is optimized?

Reply:
![image](https://github.com/iotaledger/iota-sdk/assets/20909273/a8c83cbe-27e6-4795-8d98-dc4156c6814d)  
This is a source code I used:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=258ad1b9427a99d8c434a6797e28b5a6

Reply:
I'm not sure how convincing this is, because I'm not familiar with benchmarking using criterion. It seems odd to me that `Into<Vec<_>>` could ever be slower than the collect. At best they should be identical, and at worst the collect will be slower because it is not optimized.

Reply:
I wouldn't pay attention to these small differences. This is probably a matter of order of execution. Ok, so we could investigate this another time then.